src/pkg/zoci/copier.go

```diff
@@ -5,26 +5,38 @@
 package zoci
 
 import (
+	"bytes"
 	"context"
 	"fmt"
 
+	"github.com/defenseunicorns/zarf/src/config"
 	"github.com/defenseunicorns/zarf/src/pkg/message"
 	"github.com/defenseunicorns/zarf/src/pkg/oci"
 	"github.com/defenseunicorns/zarf/src/pkg/utils/helpers"
 	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"oras.land/oras-go/v2/content"
 )
 
 // CopyPackage copies a zarf package from one OCI registry to another
-// TODO (@AustinAbro321) we should see if we should also copy the manifests for a complete copy in this function
-// Either through ORAS or additional logic found in callers of this function
 func CopyPackage(ctx context.Context, src *Remote, dst *Remote, include func(d ocispec.Descriptor) bool, concurrency int) error {
 
-	srcRoot, err := src.FetchRoot(ctx)
+	arch := config.GetArch()
+	pkg, err := src.FetchZarfYAML(ctx)
 	if err != nil {
 		return err
 	}
-	layers := helpers.Filter(srcRoot.Layers, include)
-	layers = append(layers, srcRoot.Config)
+
+	// ensure cli arch matches package arch
+	if pkg.Build.Architecture != arch {
+		return fmt.Errorf("architecture mismatch (specified: %q, found %q)", arch, pkg.Build.Architecture)
+	}
```

Comment:
I believe we can still get rid of the check here, but of note is that this check is the src `pkg.Build.Architecture`. However I still think this doesn't matter as ultimately the source architecture is also getting grabbed by the config.arch since the refactor. We may need to leave this in for backwards compatibility? I will run some tests and see if I can break it without this block